### PR TITLE
Bitmart network duplicates fix

### PR DESCRIPTION
**Pull request by clementjuventin**

### CODE REVIEW

File: ts/src/bitmart.ts [hunk 1]
```diff
@@ -2554,26 +2643,17 @@ export default class bitmart extends Exchange {
         this.checkAddress (address);
         await this.loadMarkets ();
         const currency = this.currency (code);
+        const ecid = this.safeString (params, 'network', 'ERC20');
+        const currencyId = await this.mapToCurrencyId (code, ecid);
         const request = {
-            'currency': currency['id'],
+            'currency': currencyId,
```
Comment (NeOMakinG) [inline ts/src/bitmart.ts:2649]: This is so simpler and logic!

**Review by NeOMakinG — APPROVED** [review]
Perfect 🍬

File: package.json [hunk 2]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "@cedelabs/ccxt",
-  "version": "4.0.107",
+  "version": "4.0.108",
```
Comment (NeOMakinG) [inline package.json:3]:
```suggestion
  "version": "4.0.109",
```

**Review by NeOMakinG — CHANGES_REQUESTED** [review]
Blocking for the conflicts, 4.0.108 has been deployed already

**Review by NeOMakinG — APPROVED** [review]
Fixed conflicts 🍬

File: ts/src/bitmart.ts [hunk 3]
```diff
@@ -411,6 +411,32 @@ export default class bitmart extends Exchange {
                 'defaultNetworks': {
                     'USDT': 'ERC20',
                 },
+                'networksById': {
+                    'SOLANA': 'SPL',
```
Comment (neeeekitos) [inline ts/src/bitmart.ts:415]:
```suggestion
                    'SOLANA': 'SOL',
```

In the mapping file, you specified `SOL` and not `SPL`

File: ts/src/bitmart.ts [hunk 4]
```diff
@@ -2484,22 +2583,12 @@ export default class bitmart extends Exchange {
          * @returns {object} an [address structure]{@link https://github.com/ccxt/ccxt/wiki/Manual#address-structure}
          */
         await this.loadMarkets ();
-        const currency = this.currency (code);
+        const ecid = this.safeString (params, 'network');
+        const currency = await this.mapToCurrencyId (code, ecid);
```
Comment (neeeekitos) [inline ts/src/bitmart.ts:2587]:
```suggestion
        const currencyId = await this.mapToCurrencyId (code, ecid);
```

**Review by neeeekitos — COMMENTED** [review]
@clementjuventin good job!

File: ts/src/bitmart.ts (hunk 4 above)
Comment (clementjuventin) [reply ts/src/bitmart.ts:2587]: yes

File: ts/src/bitmart.ts (hunk 3 above)
Comment (clementjuventin) [reply ts/src/bitmart.ts:415]: yes

**Pull request merged**